proto/minder/v1/minder.proto

```diff
@@ -863,9 +864,16 @@ message RuleEvaluationStatus {
     optional google.protobuf.Timestamp remediation_last_updated = 11;
     // remediation_details is the description of the remediation attempt if any
     string remediation_details = 12;
+    // rule_type_name is the name of the rule
+    string rule_type_name = 13;
+    // rule_description_name is the name to describe the rule
+    string rule_description_name = 14;
 }
 
 message GetProfileStatusByNameRequest {
+    reserved 5;
```

Comment:
I can update that, but any particular reason? I reserved them in the following format:
```
reserved 5, 6;
reserved abc, xyz;

// ... abc = 5;
// ... xyz = 6;
```
From: https://protobuf.dev/programming-guides/proto3/#deleting

> You should also reserve the field name to allow JSON and TextFormat encodings of your message to continue to parse.

Comments would also work, but it's correctness would depend on the reviewer, whereas using `reserved` keyword, the correctness would be enforced by the compiler.

Reply:
It's probably a good idea for us to reserve the names as well.  In a previous job, we only used the binary (tag number) format, so it was fine to re-use names (and sometimes desirable, as a "good name" got used for something earlier).  But I know that e.g. @AGMETEOR is using the JSON API, so we probably shouldn't change the meaning of the string names, either.

Reply:
I agree that this would break things for JSON APIS. But `RuleEvaluationStatus` has a field named `rule_name`, which is the `rule_type`. And with the introduction of `rule_name`, returning the `rule_type` in the `rule_name` field can cause a lot of unintentional errors. So, renaming would break compatibility, but it's necessary in this case. 

We can add this to release notes to make sure folks know about this change. Ideally, introducing a schema change should bump the version as well, and to serve both versions, we should have some conversion logic (taking inspiration from CRDs).

Reply:
Yeah -- we haven't _yet_ had a big enough change that we've needed to introduce API versioning.  With proto (at least) API versioning can be a bit of a pain - a bunch of copying fields that isn't clearly automateable, though Go interfaces might help over C++ or Java.